src/platform/magento2/order/orderProcessor.js

```diff
@@ -0,0 +1,302 @@
+    } else if (serverItem.qty !== clientItem.qty) {
+      logger.info(THREAD_ID + '< Wrong qty for ' + clientItem.sku, clientItem.qty, serverItem.qty)
+      syncPromises.push(api.cart.update(null, cartId, { // use magento API
+        sku: clientItem.parentSku && config.cart.setConfigurableProductOptions ? clientItem.parentSku : clientItem.sku,
+        qty: clientItem.qty,
+        product_option: clientItem.product_option,
+        item_id: serverItem.item_id,
+        quote_id: cartId
+      }, isThisAuthOrder));
+    } else {
+      // here we need just update local item_id
+      logger.info(THREAD_ID + '< Server and client items synced for ' + clientItem.sku);
+    }
+  }
+
+  for (const serverItem of serverItems) {
+    if (serverItem) {
+      const clientItem = clientItems.find(itm => productsEquals(itm, serverItem));
+
+      if (!clientItem) {
+        logger.info(THREAD_ID + '< No client item for ' + serverItem.sku + ', removing from server cart') // use magento API
+        syncPromises.push(api.cart.delete(null, cartId, { // delete server side item if not present if client's cart
+          sku: serverItem.sku,
+          item_id: serverItem.item_id
+        }, isThisAuthOrder))
+      }
+    }
+  }
+
+  const results = await Promise.all(syncPromises);
+  updateStep(job);
+  logger.info(THREAD_ID + '< Server cart in sync')
+  logger.debug(THREAD_ID + results);
+}
+
+async function placeOrder ({ orderData, shippingInformation, job }) {
+  const {
+    cart_id: cartId
+  } = orderData;
+
+  var {
+    payment_methods: [
+      {
+        code: paymentMethod
+      }
+    ]
+  } = shippingInformation;
+  const THREAD_ID = orderUtilities.getThreadId(job);
+
+  logger.info(`${THREAD_ID} Place order`);
+  const api = apiUtilities.getApi(order.store_code);
+  const orderId = await api.cart.order(null, cartId, {
+    'paymentMethod': {
+      'method': paymentMethod,
+      'additional_data': orderData.addressInformation.payment_method_additional
+    }}, orderUtilities.isThisAuthOrder(orderData));
+
+  updateStep(job);
+  logger.info(`${THREAD_ID} [OK] Order placed with ORDER ID`, orderId);
+  logger.debug(`${THREAD_ID} [OK] Order placed with ORDER ID`, orderId);
+
+  if (orderData.order_id) {
+    redisClient.set('order$$id$$' + orderData.order_id, JSON.stringify({
+      platform_order_id: orderId,
+      transmited: true,
+      transmited_at: new Date(),
+      platform: 'magento2',
+      order: orderData
+    }));
+  }
+
+  return orderId;
+}
+
+/**
+ * Prepare and send billing and shipping address to Magento
+ * @param orderData
+ * @param job
+ * @param done
+ * @returns {Promise<*>}
+ */
+async function assignAddresses ({ orderData, job, done }) {
+  const {
+    cart_id: cartId,
+    addressInformation: {
+      billingAddress: billingAddr,
+      shippingAddress: shippingAddr
+    }
+  } = orderData;
+  const api = apiUtilities.getApi(orderData.store_code);
+
+  try {
+    var countryList = await api.directory.countries();
```

Comment:
no vars

Reply:
I would use `var` in this case, because it's has functional scope and it's also used later on in function, not only in `try..catch ` block